sdv/single_table/copulas.py

```diff
@@ -141,3 +147,164 @@ def _sample(self, num_rows, conditions=None):
+                break
+
+            matrix = matrix - matrix * identity + identity
+
+            max_value = np.abs(np.abs(matrix).max())
+            if max_value > 1:
+                matrix /= max_value
+
+            eigenvalues, eigenvectors = scipy.linalg.eigh(matrix)
+            negative = eigenvalues < 0
+            iterations += 1
+
+        return matrix
+
+    @classmethod
+    def _rebuild_correlation_matrix(cls, triangular_covariance):
+        """Rebuild a valid correlation matrix from its lower half triangle.
+
+        The input of this function is a list of lists of floats of size 1, 2, 3...n-1:
+
+           [[c_{2,1}], [c_{3,1}, c_{3,2}], ..., [c_{n,1},...,c_{n,n-1}]]
+
+        Corresponding to the values from the lower half of the original correlation matrix,
+        **excluding** the diagonal.
+
+        The output is the complete correlation matrix reconstructed using the given values
+        and scaled to the :math:`[-1, 1]` range if necessary.
+
+        Args:
+            triangle_covariange (list[list[float]]):
+                A list that contains lists of floats of size 1, 2, 3... up to ``n-1``,
+                where ``n`` is the size of the target covariance matrix.
+
+        Returns:
+            numpy.ndarray:
+                rebuilt correlation matrix.
+        """
+        zero = [0.0]
+        size = len(triangular_covariance) + 1
+        left = np.zeros((size, size))
+        right = np.zeros((size, size))
+        for idx, values in enumerate(triangular_covariance):
+            values = values + zero * (size - idx - 1)
+            left[idx + 1, :] = values
+            right[:, idx + 1] = values
+
+        correlation = left + right
+        max_value = np.abs(correlation).max()
+        if max_value > 1:
+            correlation /= max_value
+
+        correlation += np.identity(size)
+
+        return cls._get_nearest_correlation_matrix(correlation).tolist()
+
+    def _rebuild_gaussian_copula(self, model_parameters):
+        """Rebuild the model params to recreate a Gaussian Multivariate instance.
+
+        Args:
+            model_parameters (dict):
+                Sampled and reestructured model parameters.
+
+        Returns:
+            dict:
+                Model parameters ready to recreate the model.
+        """
+        columns = []
+        univariates = []
+        for column, univariate in model_parameters['univariates'].items():
+            columns.append(column)
+            univariate['type'] = self._numerical_distributions.get(column, 'beta')
+            if 'scale' in univariate:
+                univariate['scale'] = max(0, univariate['scale'])
+
+            univariates.append(univariate)
+
+        model_parameters['univariates'] = univariates
+        model_parameters['columns'] = columns
+
+        covariance = model_parameters.get('covariance')
+        if covariance:
+            model_parameters['covariance'] = self._rebuild_correlation_matrix(covariance)
```

Comment:
I'm confused here, why does the `_rebuild_correlation_matrix` take in a covariance matrix, and returns one as well (or at least is stored in a parameter called covariance).

Reply:
I think this comes from `Copulas`, there is a deprecation warning for renaiming 'covariance' to 'correlation', but I can't recall why. 
```
./copulas/multivariate/gaussian.py:        warnings.warn('`covariance` will be renamed to `correlation` in v0.4.0',
```